Earley-0.13 tests fail to build

```
tests/Main.hs:14:1: error:
    Could not find module ‘UnbalancedPars’
    Use -v to see a list of the files searched for.
   |
14 | import qualified UnbalancedPars
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Upperbounding Earley. CC @massysett

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -5055,7 +5055,6 @@ expected-test-failures:
     - hpack-dhall # https://github.com/BlockScope/hpack-dhall/issues/25
 
     # Compilation failures
-    - Earley # https://github.com/commercialhaskell/stackage/issues/4193
     - ListLike # No issue tracker, e-mail sent to maintainer
     - amazonka-core # https://github.com/brendanhay/amazonka/issues/397
     - async-timer # https://github.com/mtesseract/async-timer/issues/8

